api/src/main/java/uk/gov/dhsc/htbhf/dwp/service/EligibilityService.java

```diff
@@ -3,23 +3,82 @@
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
+import uk.gov.dhsc.htbhf.dwp.entity.Child;
+import uk.gov.dhsc.htbhf.dwp.entity.legacy.LegacyHousehold;
+import uk.gov.dhsc.htbhf.dwp.entity.uc.UCHousehold;
 import uk.gov.dhsc.htbhf.dwp.model.EligibilityRequest;
 import uk.gov.dhsc.htbhf.dwp.model.EligibilityResponse;
+import uk.gov.dhsc.htbhf.dwp.repository.LegacyHouseholdRepository;
+import uk.gov.dhsc.htbhf.dwp.repository.UCHouseholdRepository;
+
+import java.time.LocalDate;
+import java.util.Optional;
+import java.util.Set;
 
 @Service
 public class EligibilityService {
 
     private static final String ENDPOINT = "/v1/dwp/benefits";
     private final String uri;
     private final RestTemplate restTemplate;
+    private final UCHouseholdRepository ucHouseholdRepository;
+    private final LegacyHouseholdRepository legacyHouseholdRepository;
 
-    public EligibilityService(@Value("${dwp.base-uri}") String baseUri, RestTemplate restTemplate) {
+    public EligibilityService(@Value("${dwp.base-uri}") String baseUri,
+                              RestTemplate restTemplate,
+                              UCHouseholdRepository ucHouseholdRepository,
+                              LegacyHouseholdRepository legacyHouseholdRepository) {
         this.uri = baseUri + ENDPOINT;
         this.restTemplate = restTemplate;
+        this.ucHouseholdRepository = ucHouseholdRepository;
+        this.legacyHouseholdRepository = legacyHouseholdRepository;
     }
 
     public EligibilityResponse checkEligibility(EligibilityRequest eligibilityRequest) {
+        String nino = eligibilityRequest.getPerson().getNino();
+        Optional<UCHousehold> ucHousehold = ucHouseholdRepository.findHouseholdByAdultWithNino(nino);
+        if (ucHousehold.isPresent()) {
+            return createEligibilityResponse(ucHousehold.get());
+        }
+
+        Optional<LegacyHousehold> legacyHousehold = legacyHouseholdRepository.findHouseholdByAdultWithNino(nino);
+        if (legacyHousehold.isPresent()) {
+            return createEligibilityResponse(legacyHousehold.get());
+        }
+
         var response = restTemplate.postForEntity(uri, eligibilityRequest, EligibilityResponse.class);
         return response.getBody();
     }
+
+    private EligibilityResponse createEligibilityResponse(LegacyHousehold household) {
```

Comment:
It looks like these methods should be refactored out into an `EligibilityResponseBuilder` and not live in the service, it'll be easier to test then too, your current service tests are quite lengthy.

Reply:
Done.